lib/new_relic/agent/instrumentation/roda/instrumentation.rb

```diff
@@ -0,0 +1,55 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module Roda
+    module Tracer
+      include ::NewRelic::Agent::Instrumentation::ControllerInstrumentation
+
+      def self.included(clazz)
+        clazz.extend(self)
+      end
+
+      def newrelic_middlewares
+        middlewares = [NewRelic::Rack::BrowserMonitoring]
+        if NewRelic::Rack::AgentHooks.needed?
+          middlewares << NewRelic::Rack::AgentHooks
+        end
+        middlewares
+      end
+
+      def build_rack_app_with_tracing
+        unless NewRelic::Agent.config[:disable_roda_auto_middleware]
+          newrelic_middlewares.each do |middleware_class|
+            self.use middleware_class
+          end
+        end
+        yield
+      end
+
+      # Roda makes use of Rack, so we can get params from the request object
+      def rack_request_params
+        begin
+          @_request.params
+        rescue => e
+          NewRelic::Agent.logger.debug('Failed to get params from Rack request.', e)
+          nil
```

Comment:
```suggestion
          NewRelic::EmptyHash
```

from what I can tell, the result here winds up getting passed in as the `params` (second) argument to `NewRelic::Agent::ParameterFiltering.filter_rack_file_data`, which in turns invokes `#inject` on it. `#inject` won't work on `nil`...

Reply:
Ah, I missed the `||` you refactored out with [160a64b](https://github.com/newrelic/newrelic-ruby-agent/pull/2144/commits/160a64b6fbbc3f6dfb13af9c2770c2f76287a29b). That explains it. Sorry about that. I think I remember seeing it in pairing but forgot when it came to the PR. I think the refactor looks better, so we should keep it if we can get the tests in line.